engineio/src/client/socket.rs

```diff
@@ -32,19 +38,72 @@ impl SocketBuilder {
             headers: None,
             tls_config: None,
             handshake: None,
+            on_close: OptionalCallback::default(),
```

Comment:
Didn't you plan on putting those in a map? Or is that something that happens in a separate pr?

Reply:
Ah wait that's the builder, nevermind!

Reply:
I attempted to put them both in a map and struct, opted to go for the way it was.
Map issues:
- The generic type varies between the callbacks, could use a dyn object to try to capture this, that has it's own issues
Struct issues:
- No real value add
- In order to retrieve the values from the struct the function name to get the callback out (on_close or close) taking no parameters looks more (from a socketio background) like a function that would call the events or handlers, not simple getters
- Difficult to get type checking on enclosures to resolve correctly due to lack of trait aliases in rust

Reply:
Ah oke I See! Didn't think about the generic type parameter. The type parameter could be refactored into an enum, but I'm not sure if that brings some enhancement or just makes things more difficult. I think the current way is pretty straight forward.